### PR TITLE
Refresh plugin for June 2023

### PR DESCRIPTION
**Pull request by basil**
Make tests more stable and robust.

See also discussions in #512 

### Testing done

`mvn clean verify`

### Proposed upgrade guidelines

N/A

### Localizations

N/N

### Submitter checklist

- ~~[ ] The Jira / Github issue, if it exists, is well-described.~~
- [x] The changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developers, depending on the change) and are in the imperative mood (see [examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)).
  - The changelog generator for plugins uses the **pull request title as the changelog entry**.
  - Fill in the **Proposed upgrade guidelines** section only if there are breaking changes or changes that may require extra steps from users during the upgrade.
- [x] There is automated testing or an explanation that explains why this change has no tests.
- ~~[ ] New public functions for internal use only are annotated with `@NoExternalUse`. In case it is used by non java code the `Used by {@code <panel>.jelly}` Javadocs are annotated.~~
- ~~[ ] New or substantially changed JavaScript is not defined inline and does not call `eval` to ease the future introduction of Content Security Policy (CSP) directives (see [documentation](https://www.jenkins.io/doc/developer/security/csp/)).~~
- ~~[ ] For new APIs and extension points, there is a link to at least one consumer.~~
- ~~[ ] Any localizations are transferred to *.properties files.~~
- ~~[ ] Changes in the interface are documented also as [examples](src/doc/examples/readme.md).~~

### CODE REVIEW

**Comment by basil** [comment]
Used my admin powers to replay the build with the `Jenkinsfile` changes here: https://ci.jenkins.io/job/Plugins/job/lockable-resources-plugin/job/PR-519/3/

**Comment by basil** [comment]
CC @mPokornyETM

**Review by mPokornyETM — APPROVED** [review]
LGTM

File: Jenkinsfile [hunk 1]
```diff
@@ -3,11 +3,6 @@
...
  */
 buildPlugin(useContainerAgent: true, configurations: [
-  // Test the common case (i.e., a recent LTS release) on both Linux and Windows.
-  [ platform: 'linux', jdk: '11' ],
-  [ platform: 'windows', jdk: '11'],
-
-  // Test the bleeding edge of the compatibility spectrum (i.e., the latest supported Java runtime).
-  // see also https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/
-  [ platform: 'linux', jdk: '17', jenkins: '2.387.1' ],
+  [platform: 'linux', jdk: 17],
```
[the remark this answers is not in the document]
Comment (basil) [reply Jenkinsfile:6]: To conform to the archetype.

**Pull request merged**